Sources/Tokenizer/Layout/Constraints/Constraint.swift

```diff
@@ -101,6 +108,209 @@ public struct Constraint {
+            return "\(viewName).traits.orientation(\(orientation.traitsParameter))"
+        case .number(let number):
+            return String(number)
+        case .dimensionType(let dimensionType):
+            return "\(viewName).traits.topViewSize(\(dimensionType.traitsParameter))"
+        default:
+            return ""
+        }
+    }
+
+    // MARK: - XML condition generators
+    public func generateXMLCondition(condition: Condition) -> String {
+        switch condition {
+        case .statement(let statement):
+            return generateXMLCondition(statement: statement)
+        case .unary(let operation, let condition):
+            return "\(operation.xmlRepresentation)\(generateXMLCondition(condition: condition))"
+        case .binary(let operation, let lhsCondition, let rhsCondition):
+            return [
+                generateXMLCondition(condition: lhsCondition),
+                operation.xmlRepresentation,
+                generateXMLCondition(condition: rhsCondition),
+            ].joined(separator: " ")
+        }
+    }
+
+    public func generateXMLCondition(statement: ConditionStatement) -> String {
+        switch statement {
+        case .trueStatement:
+            return "true"
+        case .falseStatement:
+            return "false"
+        case .sizeClass(let sizeClassType, let viewInterfaceSize):
+            return "\(sizeClassType.description) == \(viewInterfaceSize.description)"
+        case .interfaceIdiom(let interfaceIdiom):
+            return interfaceIdiom.description
+        case .orientation(let orientation):
+            return orientation.description
+        case .number(let number):
+            return String(number)
+        case .dimensionType(let dimensionType):
+            return dimensionType.description
+        default:
+            return ""
+        }
+    }
+}
+
+struct ConditionError: Error {
+    let message: String
+
+    init(_ message: String) {
+        self.message = message
+    }
+}
+
+extension ConditionBinaryOperation {
+    var swiftRepresentation: String {
+        switch self {
+        case .equal:
+            return "=="
+        case .and:
+            return "&&"
+        case .or:
+            return "||"
+        case .less:
+            return "<"
+        case .greater:
+            return ">"
+        }
+    }
+
+    var xmlRepresentation: String {
+        switch self {
+        case .equal:
+            return swiftRepresentation
+        case .and:
+            return "and"
+        case .or:
+            return "or"
+        case .less:
+            return ":lt"
+        case .greater:
+            return ":gt"
+        }
+    }
+}
+
+extension ConditionUnaryOperation {
+    var swiftRepresentation: String {
+        switch self {
+        case .none:
+            return ""
+        case .negation:
+            return "!"
+        }
+    }
+
+    var xmlRepresentation: String {
+        switch self {
```

Comment:
Why not return `swiftRepresentation` without switching?

Reply:
It's premature change for future maintainer when we wants to add new unary operator. I guess we can have a dumb repeater for now.